### PR TITLE
Add components from prototype to this project

### PR DESCRIPTION
**Pull request by martypowell**
- Added components to the project from the prototype
- Updated demo to include filter list, instead of placeholder code
- Ensure Filter List works as desired based on the demo
- Changed axios to use window.fetch (with polyfill) in order to work better with packaging
- Changed `useQuery` to `useInfiniteQuery` per the react-query [docs](https://github.com/tannerlinsley/react-query#load-more--infinite-scroll-with-useinfinitequery)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,5 +37,12 @@
   },
   "files": [
     "dist"
-  ]
+  ],
+  "dependencies": {
+    "@baltimorecounty/dotgov-components": "^1.0.0",
+    "node-fetch": "^2.6.0",
```
Comment (martypowell) [inline package.json]: lines 43 and 44 can be removed

Comment (martypowell) [reply package.json]: actually only 43, removed

File: rollup.config.js [hunk 2]
```diff
@@ -8,7 +9,11 @@ export default [
     output: [{ file: pkg.demo, format: "cjs" }],
     external: ["react", "react-dom"],
     plugins: [
+      commonjs({
```
Comment (martypowell) [inline rollup.config.js]: can we set this to one variable?

Comment (martypowell) [reply rollup.config.js]: this was possible, updated

File: src/components/FilterList.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import React, { useState } from "react";
+
+import ApiList from "./ApiList.jsx";
+import Filters from "./Filters.jsx";
+import { UpdateQueryString } from "../common/Filters";
+
+const FilterList = ({
```
Comment (martypowell) [inline src/components/FilterList.jsx:8]: we should probably add proptypes for at least this component.

Comment (martypowell) [reply src/components/FilterList.jsx:8]: added for this component

**Review by tmccoy529 — DISMISSED** [review]

File: demo/Demo.jsx [hunk 4]
```diff
@@ -1,17 +1,37 @@
...
-      <Paragraph>This is some really meaningful text.</Paragraph>
-      <Paragraph>This is some really meaningful text.</Paragraph>
-      <Paragraph>This is some really meaningful text.</Paragraph>
-      <Paragraph>This is some really meaningful text.</Paragraph>
-      <Paragraph>This is some really meaningful text.</Paragraph>
-      <Paragraph>This is some really meaningful text.</Paragraph>
+      <FilterList
+        title="Baltimore County News"
+        filters={filters}
+        apiEndpoint="https://structuredcontentdev.bcg.ad.bcgov.us/api/news"
```
Comment (martypowell) [inline demo/Demo.jsx]: for demo purposes maybe there is some public api we can use?

File: src/components/ApiList.jsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import "promise-polyfill/src/polyfill";
+import "whatwg-fetch";
+
+import React from "react";
+import RecordsMessage from "./RecordsMessage";
+import { useInfiniteQuery } from "react-query";
+
+const fetchList = (key, { endpoint }, loadMoreEndpoint) =>
```
Comment (martypowell) [inline src/components/ApiList.jsx:17]: a summary here might be good, especially for the endpoint check.

**Comment by martypowell** [comment]
note to self, this would be a good opportunity to create a readme, as this will probably be a prerelease 0.1.0 so that we can work out the bugs, but the expectation is this isn't far from v1.0.0

**Comment by martypowell** [comment]
> note to self, this would be a good opportunity to create a readme, as this will probably be a prerelease 0.1.0 so that we can work out the bugs, but the expectation is this isn't far from v1.0.0

this was added.

File: src/components/ApiList.jsx (hunk 5 above)
Comment (martypowell) [reply src/components/ApiList.jsx:17]: this was added.

File: demo/Demo.jsx (hunk 4 above)
Comment (martypowell) [reply demo/Demo.jsx]: added placeholder for this.

**Review by tmccoy529 — APPROVED** [review]

**Pull request merged**